src/runtime_src/core/tools/xbmgmt2/OO_Config.cpp

```diff
@@ -19,32 +19,257 @@
 #include "tools/common/XBUtilities.h"
 namespace XBU = XBUtilities;
 
+#include "core/common/device.h"
+#include "core/common/system.h"
+#include "core/common/error.h"
+#include "core/common/utils.h"
+#include "core/common/query_requests.h"
+
 // 3rd Party Library - Include Files
+#include <boost/format.hpp>
+#include <boost/program_options.hpp>
+namespace po = boost::program_options;
 
 // System - Include Files
 #include <iostream>
 
+
+// =============================================================================
+
+// ------ L O C A L   F U N C T I O N S ---------------------------------------
+
+namespace {
+
+enum class configType {
+    config_security = 0,
+    config_clk_scaling,
+    config_cs_threshold_power_override,
+};
+
+enum class memType {
+    unknown= 0,
```

Comment:
The individual enumeration values should have a common per-appended value to indicate what enumeration this value is from and to prevent name collisions.  Some possible examples could be:
memType_unknown 
```
MT_unknown
memt_unknown
MemT_unknown
```

Reply:
The beauty of enum classes is that there is no name collision.   Let's all use enum class so that enum variables are scoped to the class and let's always use lower case enum variable names  that are succinctly named.